crates/vm/src/arch/vm.rs

```diff
@@ -752,7 +686,14 @@ where
             executor.execute_from_state(&mut exec_state)
         })?;
 
-        check_exit_code(exec_state.exit_code)?;
+        match exec_state.exit_code {
```

Comment:
why did you have to get rid of `check_exit_code`?

Reply:
i inlined it because in cases when you're running for a fixed number of cycles, you don't want to throw a `DidNotTerminate` error when `exit_code` is `None`. so the name `check_exit_code` seemed a bit confusing

Reply:
added it back as `check_termination`